src/GNDStk/Node/test/subscript.test.cpp

```diff
@@ -1,26 +1,691 @@
+            // v will be a detail::meta_ref<Node,CONST=true,version_t>,
+            // where version_t is our custom type defined above
+            auto v = c(reactionSuite,styles,evaluated)
+               [meta_t<version_t>("version")];
+
+            // for const, assignment isn't available; neither of these compile:
+            // v = version_t(8,0,3);
+            // v = "8.0.3";
+
+            // v's type, as created by [] and described above, isn't just
+            // an rvalue version_t as we'd have gotten from (). Instead, it's
+            // something more intelligent. Here, we see that it can convert
+            // (in the C++ sense, not in GNDStk's convert() sense) to either
+            // a version_t or to a std::string. A return from (), not from []
+            // as we had above, wouldn't have been able to do that.
+            const version_t v_new = v; // v can become a version_t
+            const std::string str = v; // v can become a std::string
+
+            CHECK(v_new.major == 8);
+            CHECK(v_new.minor == 0);
+            CHECK(v_new.patch == 3);
+            CHECK(str == "8.0.3");
+         }
+
+         // ------------
+         // NON-CONST
+         // ------------
+
+         THEN("non-const works as expected") {
+            // as for the const case, above, but assignment is now available
+            auto v = n(reactionSuite,styles,evaluated)
+               [meta_t<version_t>("version")];
+
+            // Assignment is available now, because the above came from our
+            // non-const node. The following illustrates the real point of
+            // using the [] operator: to get an lvalue:
+            v = version_t(8,0,3);
+            v = "8.0.3";
+
+            // as before
+            const version_t v_new = v; // v can become a version_t
+            const std::string str = v; // v can become a std::string
+
+            CHECK(v_new.major == 8);
+            CHECK(v_new.minor == 0);
+            CHECK(v_new.patch == 3);
+            CHECK(str == "8.0.3");
+
+            // Let's exercise [] obtained from a non-const node, not so much
+            // for the sake of more tests, but to illustrate []'s main purpose
+            // as something that's usable (when applied to non-const nodes
+            // anyway) as an lvalue.
+
+            auto &EvaluatedNode = n(reactionSuite,styles,evaluated);
+            const meta_t<version_t> myversion("version");
+
+            // Actually writes into the tree...
+            EvaluatedNode[myversion] = "123.456.789";
+            CHECK(tree(reactionSuite,styles,evaluated,basic::version) ==
+                  "123.456.789");
+            // can still use as rvalue, and implicitly converts to string...
+            std::string foo = EvaluatedNode[myversion];
+            CHECK(foo == "123.456.789");
+
+            // Also actually writes into the tree...
+            EvaluatedNode[myversion] = version_t(12,34,56);
```

Comment:
I assume the prerequisite for this to work is to have an ostream operator>> defined for version_t?

Builtin types like double, int, etc. have this by default. We should discuss precision in floating points at some point.

Reply:
Re: version_t, the prerequisite is that it have the usual GNDStk `convert()` business. However, my built-in `convert()` functions for `meta_t` eventually fall back on stream I/O if all else fails.

Reply:
Let's come up with a short example in which we write a couple of small sections of GNDStk data, and want to set precision individually in each one.